study.md

```diff
@@ -21,18 +21,18 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+<!-- I was confused about hoisting.  How do the variables know to move up before the alert or return statement. -->
```

Comment:
Here's a good article on JavaScript's variable hoisting.  https://en.wikipedia.org/wiki/JavaScript_syntax#Scoping_and_hoisting

I wrote a small script as an example.  Let me know if it's unclear.
```javascript
// This will output, "[Function: sayHello]".
console.log(sayHello);

// This will work and output, "Hello".
sayHello();

// This will output, "undefined" since the declaration of "sayHi" has been
// hoisted to the top, but its definition has not been.
console.log(sayHi);

// This will throw an error for the reason above.  Since sayHi is undefined,
// it's trying to execute "undefined" as if it were a function.
// sayHi();

// This gets hoisted to the top.
function sayHello() {
  console.log('Hello.');
}

// "var sayHi" gets hoisted to the top, but its assignment to function() { ... }
// does not.  Variable declarations using "const" and "let" don't get hoisted.
const sayHi = function() {
  console.log('Hi.');
};

// This will output, "[Function: sayHi]", since it's after the definition of the
// function.
console.log(sayHi);

// This will output, "Hi".
sayHi();
```